src/hotspot/share/gc/g1/g1FullGCMarker.cpp

```diff
@@ -42,7 +44,13 @@ G1FullGCMarker::G1FullGCMarker(G1FullCollector* collector,
     _mark_closure(worker_id, this, G1CollectedHeap::heap()->ref_processor_stw()),
     _verify_closure(VerifyOption_G1UseFullMarking),
     _stack_closure(this),
-    _cld_closure(mark_closure(), ClassLoaderData::_claim_strong) {
+    _cld_closure(mark_closure(), ClassLoaderData::_claim_strong),
+    // cache size is big enough that not increase pause during marking
+    // by avoiding hit misses as so as possible
+    _mark_region_cache(mark_stats, round_up_power_of_2(G1CollectedHeap::heap()->max_regions())) {
```

Comment:
The suggested cache size is way too huge for large heaps (10k's of regions) for no gain. Unless you have good numbers, please use the same size as for the concurrent mark stats cache (fixed 1024).
Feel free to move `G1CMTask::RegionMarkStatsCacheSize` to `G1RegionMarkStatsCache` and make it a public static constant there.

Reply:
already done in pr 2966